src/LayoutSdk/LayoutSdk.proj

```diff
@@ -0,0 +1,29 @@
+<!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the MIT license. See License.txt in the project root for full license information. -->
+<Project Sdk="Microsoft.NET.Sdk">
+  <Target Name="LayoutSdk">
+    <ItemGroup>
+      <Stage0SdkPaths Include="$([System.IO.Directory]::GetDirectories(&quot;$(_DotNetRoot)sdk\&quot;))"/>
+    </ItemGroup>
+    <PropertyGroup>
+      <!-- Reverse list, and put in property -->
+      <Stage0SdkPaths>@(Stage0SdkPaths->Reverse())</Stage0SdkPaths>
+      <!-- Get first path in reversed list, which should hopefully be the latest version -->
+      <Stage0SdkPath>$(Stage0SdkPaths.Split($([System.Convert]::ToString(";").ToCharArray())).GetValue($([System.Convert]::ToInt32(0))))</Stage0SdkPath>
+      <Stage0SdkVersion>$([System.IO.Path]::GetFileName('$(Stage0SdkPath)'))</Stage0SdkVersion>
```

Comment:
I tried to do this for the CLI and I failed miserably. Good to know it is possible if you know enough msbuildfoo.

Reply:
I always ended up with NuGetFallbackFolder getting in the way for this algorithm. Are we saved by not installing < 3.0 SDKs?

Reply:
> `$([System.Convert]::ToString(";").ToCharArray())`

Is this necessary? We have Split(';') elsewhere.

Reply:
> which should hopefully be the latest version

I don't think GetDirectories() has this order guaranteed by anything.

Reply:
This is the correct algorithm:
 https://github.com/dotnet/sdk/blob/3659227954921dc77de97f15215e96e6027468d0/src/Tests/Microsoft.NET.TestFramework/ToolsetInfo.cs#L100-L106

Reply:
Maybe Exec `$(_DotnetRoot)` --version?

Reply:
I've switched to using the `Exec` task on `dotnet --version`.  This can still be wrong locally if you switch to a branch with an earlier stage 0, but that's a problem we have anyway in other places, and when this runs on CI it should be with a clean enlistment.